tests/extension/handler/utils.test.ts

```diff
@@ -87,3 +85,13 @@ describe('getSuggestedProjectName', () => {
         expect(window.showErrorMessage).toBeCalledTimes(1)
     })
 })
+
+describe('parseParams', () => {
+  it('should parse params to be used safely', () => {
+    const usp = new URLSearchParams('fileToOpen=foo;bar loo&repository=git@github.com/org/project;foo bar.git')
+    expect(parseParams(usp)).toEqual({
+      fileToOpen: 'foo%3Bbar%20loo',
+      repository: 'git@github.com/org/project%3Bfoo%20bar.git'
```

Comment:
@christian-bromann is a URL encoded repository ref not an invalid ref?

Reply:
I am not sure what you mean by that. This would fail later because the repo is not found this way, however it prevents injecting malicious stuff into the terminal.